cypress/e2e/DoenetML/tagSpecific/sampleprimenumbers.cy.js

```diff
@@ -0,0 +1,2166 @@
+      expect(
+        stateVariables[
+          stateVariables[stateVariables["/p8"].replacements[0].componentName]
+            .activeChildren[0].componentName
+        ].activeChildren.length,
+      ).eq(1);
+      expect(
+        stateVariables[
+          stateVariables[stateVariables["/p9"].replacements[0].componentName]
+            .activeChildren[0].componentName
+        ].activeChildren.length,
+      ).eq(1);
+      for (let ind = 0; ind < 1; ind++) {
+        expect(
+          stateVariables[
+            stateVariables[
+              stateVariables["/p1"].activeChildren[0].componentName
+            ].activeChildren[ind].componentName
+          ].stateValues.value,
+        ).eq(samplednumbers[ind]);
+        expect(
+          stateVariables[
+            stateVariables[
+              stateVariables["/p2"].activeChildren[0].componentName
+            ].activeChildren[ind].componentName
+          ].stateValues.value,
+        ).eq(samplednumbers[ind]);
+        expect(
+          stateVariables[
+            stateVariables[
+              stateVariables["/p3"].activeChildren[0].componentName
+            ].activeChildren[ind].componentName
+          ].stateValues.value,
+        ).eq(samplednumbers[ind]);
+        expect(
+          stateVariables[
+            stateVariables[
+              stateVariables[
+                stateVariables["/p4"].replacements[0].componentName
+              ].activeChildren[0].componentName
+            ].activeChildren[ind].componentName
+          ].stateValues.value,
+        ).eq(samplednumbers[ind]);
+        expect(
+          stateVariables[
+            stateVariables[
+              stateVariables[
+                stateVariables["/p5"].replacements[0].componentName
+              ].activeChildren[0].componentName
+            ].activeChildren[ind].componentName
+          ].stateValues.value,
+        ).eq(samplednumbers[ind]);
+        expect(
+          stateVariables[
+            stateVariables[
+              stateVariables[
+                stateVariables["/p6"].replacements[0].componentName
+              ].activeChildren[0].componentName
+            ].activeChildren[ind].componentName
+          ].stateValues.value,
+        ).eq(samplednumbers[ind]);
+        expect(
+          stateVariables[
+            stateVariables[
+              stateVariables[
+                stateVariables["/p7"].replacements[0].componentName
+              ].activeChildren[0].componentName
+            ].activeChildren[ind].componentName
+          ].stateValues.value,
+        ).eq(samplednumbers[ind]);
+        expect(
+          stateVariables[
+            stateVariables[
+              stateVariables[
+                stateVariables["/p8"].replacements[0].componentName
+              ].activeChildren[0].componentName
+            ].activeChildren[ind].componentName
+          ].stateValues.value,
+        ).eq(samplednumbers[ind]);
+        expect(
+          stateVariables[
```

Comment:
Reading these can get a little dizzying, we don't have to fix it now, but a couple utility methods could probably go a long way to increasing the information density of these tests. It also wouldn't be terrible to have a central place to describe this pattern for digging into the stateVariables, its a bit hard to follow what this is even doing, which is kind of impressive for code with no branches in it.

Reply:
Looking at this a bit closer I see a bunch of this redundancy could also be reduced with just some more loops too, as the changes between many of these copy/pasted blocks is just a single string component name. I still think a utility function that describes what this code is doing wouldn't be a bad addition as well.

Reply:
I have to admit: I just wholesale copied the tests from samplerandomnumbers.cy.js and selectfromsequence.cy.js, which were written a long time ago, and made as few changes as possible, without working to hard to understand how they work.  I'm going to put these in as they are, as we should just clean up all these select/sample tests to improve their readability and redundancy, which is more than I'm up for doing now.